packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -10,47 +10,60 @@ import '../../devfs.dart';
 import '../../plugins.dart';
 import '../../project.dart';
 import '../build_system.dart';
+import '../phases.dart';
 
-/// The copying logic for flutter assets.
-class AssetBehavior extends SourceBehavior {
-  const AssetBehavior();
+class AssetsBuildPhase extends BuildPhase {
+  const AssetsBuildPhase();
 
   @override
-  List<File> inputs(Environment environment) {
+  Future<List<Target>> plan(Environment environment) async {
+    final List<Source> inputs = <Source>[
+      const Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/assets.dart'),
```

Comment:
If one input is changed, is every file re-copied?

Reply:
For now, yes. longer/medium term I need to do more research into how gn/blaze handle copy steps. Most build systems seem to have special cases for this.

Reply:
Consider filing an issue and tracking as part of the assemble project.